### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
Without the stack traces, some of the errors would say little more than "Runtime exception" and be very hard to diagnose and fix.

# What has changed
Added descriptive text to all RuntimeExceptions. Added option to print stack traces which can be toggled on to diagnose difficult problems.

# How to test?
Try toggling the setting in application.yml and set off some exceptions.

# Links
Trello: https://trello.com/c/KViVYGvP

### CODE REVIEW

**Comment by NickGrantONS** [comment]
[Improved Logging & Optional Stack traces](https://trello.com/c/KViVYGvP/1075-improved-logging-optional-stack-traces)

File: src/main/java/uk/gov/ons/census/casesvc/messaging/ActionSchedulerEventReceiver.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public void receiveMessage(ResponseManagementEvent event) {
     } else if (event.getEvent().getType() == EventTypeDTO.FIELD_CASE_SELECTED) {
       eventService.processFieldCaseSelected(event);
     } else {
-      throw new RuntimeException(); // Unexpected event type received
+      throw new RuntimeException("Unexpected event type received");
```
Comment (AdeHarrison) [inline src/main/java/uk/gov/ons/census/casesvc/messaging/ActionSchedulerEventReceiver.java]: include event.getEvent().getType() in message

File: src/main/java/uk/gov/ons/census/casesvc/service/CaseService.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public int getUniqueCaseRef() {
 
     // Check for collisions
     if (caseRepository.existsById(caseRef)) {
-      throw new RuntimeException();
+      throw new RuntimeException("Case ref collision");
```
Comment (AdeHarrison) [inline src/main/java/uk/gov/ons/census/casesvc/service/CaseService.java]: include caseRef in message

File: src/main/java/uk/gov/ons/census/casesvc/utility/QuestionnaireTypeHelper.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ public static int calculateQuestionnaireType(String treatmentCode) {
       throw new IllegalArgumentException();
     }
 
-    throw new RuntimeException(); // This code should be unreachable
+    throw new RuntimeException("Unprocessable treatment code");
```
Comment (AdeHarrison) [inline src/main/java/uk/gov/ons/census/casesvc/utility/QuestionnaireTypeHelper.java]: include treatmentCode in message

**Pull request merged**